tests/e2e/e2e_trafficsplit_test.go

```diff
@@ -0,0 +1,253 @@
+				ports:     []int{80},
+			})
+
+			// Creating trafficsplit service in K8s
+			_, err := td.CreateService(serverNamespace, trafficSplitService)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Create Traffic split with all server processes as backends
+			trafficSplit := TrafficSplitDef{
+				Name:                    trafficSplitName,
+				Namespace:               serverNamespace,
+				TrafficSplitServiceName: trafficSplitName,
+				Backends:                []TrafficSplitBackend{},
+			}
+			assignation := 100 / len(serverServices) // Spreading equitatively
+			for _, dstServer := range serverServices {
+				trafficSplit.Backends = append(trafficSplit.Backends,
+					TrafficSplitBackend{
+						Name:   dstServer,
+						Weight: assignation,
+					},
+				)
+			}
+			// Get the Traffic split structures
+			tSplit, err := td.CreateSimpleTrafficSplit(trafficSplit)
+			Expect(err).To(BeNil())
+
+			// Push them in K8s
+			_, err = td.CreateTrafficSplit(serverNamespace, tSplit)
+			Expect(err).To(BeNil())
+
+			// Test traffic
+			// Create Multiple HTTP request structure
+			requests := HTTPMultipleRequest{
+				Sources: []HTTPRequestDef{},
+			}
+			for _, ns := range clientServices {
+				pods, err := td.client.CoreV1().Pods(ns).List(context.Background(), metav1.ListOptions{})
+				Expect(err).To(BeNil())
+
+				for _, pod := range pods.Items {
+					requests.Sources = append(requests.Sources, HTTPRequestDef{
+						SourceNs:        ns,
+						SourcePod:       pod.Name,
+						SourceContainer: ns, // container_name == NS for this test
+
+						// Targeting the trafficsplit FQDN
+						Destination: fmt.Sprintf("%s.%s", trafficSplitName, serverNamespace),
+
+						HTTPUrl: "/",
+						Port:    80,
+					})
+				}
+			}
+
+			var results HTTPMultipleResults
+			var serversSeen map[string]bool = map[string]bool{} // Just counts unique servers seen
+			success := td.WaitForRepeatedSuccess(func() bool {
+				curlSuccess := true
+
+				// Get results
+				results = td.MultipleHTTPRequest(&requests)
+
+				// Print results
+				td.PrettyPrintHTTPResults(&results)
+
+				// Verify REST status code results
+				for _, ns := range results {
+					for _, podResult := range ns {
+						if podResult.Err != nil || podResult.StatusCode != 200 {
+							curlSuccess = false
+						} else {
+							// We should see pod header populated
+							dstPod, ok := podResult.Headers[HTTPHeaderName]
+							if ok {
+								// Store and mark that we have seen a response for this server pod
+								serversSeen[dstPod] = true
+							}
+						}
+					}
+				}
+				td.T.Logf("Unique servers replied %d/%d",
+					len(serversSeen), numberOfServerServices*serverReplicaSet)
+
+				// Success conditions:
+				// - All clients have been answered consecutively 5 successful HTTP requests
+				// - We have seen all servers from the traffic split reply at least once
+				return curlSuccess && (len(serversSeen) == numberOfServerServices*serverReplicaSet)
+			}, 5, 150*time.Second)
```

Comment:
please use variables/constants/comments for these values.

Reply:
I want to file an issue to parametrize those. It's on my radar.

Reply:
https://github.com/openservicemesh/osm/issues/1825